api/v1alpha2/types.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import "strings"
+
+// APIEndpoint represents a reachable Kubernetes API endpoint.
+type APIEndpoint struct {
+	// The hostname on which the API server is serving.
+	Host string `json:"host"`
+	// The port on which the API server is serving.
+	Port int `json:"port"`
+}
+
+// DOSafeName returns DigitalOcean safe name with replacing '.' and '/' to '-'
+// since DigitalOcean doesn't support naming with those character.
+func DOSafeName(name string) string {
+	r := strings.NewReplacer(".", "-", "/", "-")
+	return r.Replace(name)
+}
+
+// DOResourceStatus describes the status of a DigitalOcean resource.
+type DOResourceStatus string
+
+var (
+	// DOResourceStatusNew is the string representing a DigitalOcean resource just created and in a provisioning state.
+	DOResourceStatusNew = DOResourceStatus("new")
+	// DOResourceStatusRunning is the string representing a DigitalOcean resource already provisioned and in a active state.
+	DOResourceStatusRunning = DOResourceStatus("active")
+	// DOResourceStatusErrored is the string representing a DigitalOcean resource in a errored state.
+	DOResourceStatusErrored = DOResourceStatus("errored")
+	// DOResourceStatusOff is the string representing a DigitalOcean resource in off state.
+	DOResourceStatusOff = DOResourceStatus("off")
+	// DOResourceStatusArchine is the string representing a DigitalOcean resource in archive state.
+	DOResourceStatusArchine = DOResourceStatus("archive")
```

Comment:
whoops some typos :grin: 

yeah @xmudrii that status is relevant to DigitalOcean status. https://developers.digitalocean.com/documentation/v2/#droplets

> A status string indicating the state of the Droplet instance. This may be "new", "active", "off", or "archive".

but I am not sure when it was used, the current state `archive` status is not used. i just leave that status there. are you prefer to remove this for now?

Reply:
In this case, let's leave it. I wasn't aware that this state exists, but as it exists, let's leave it.